src/manage_sql_secinfo.c

```diff
@@ -3093,22 +3093,20 @@ update_cert_timestamp ()
                      __func__,
                      timestamp);
           g_free (timestamp);
-          return -1;
+          stamp = time(NULL);
         }
 
       timestamp[8] = '\0';
       g_debug ("%s: parsing: %s", __func__, timestamp);
       stamp = parse_feed_timestamp (timestamp);
       g_free (timestamp);
       if (stamp == 0)
-        return -1;
+        stamp = time(NULL);
```

Comment:
Does it make sense to add an else around those lines ? If the above if-condition is true, we already have freed `timestamp`.